### ISSUE TITLE
GCE: issue with either defaults or documentation for google_container_cluster/node_config/oauth_scopes

### ISSUE THREAD

**Issue by publysher** [hashicorp/terraform#3681]
The default setting for `google_container_cluster/node_config/oauth_scopes` is currently the empty list. This means that when you use the example from [the terraform documentation](https://terraform.io/docs/providers/google/r/container_cluster.html), you will get a cluster without any oauth_scopes. 

However, this means that when you run the [default example with persistent disks](https://cloud.google.com/container-engine/docs/tutorials/persistent-disk/), you will get errors like this:

```
Timed out waiting for GCE PD "mysql-disk" to attach. Retrying attach.
```

I eventually found the problem, but it is very annoying from a "quickly get up and running" point of view. 

I would suggest one of the following solutions:

1. Use the list `['https://www.googleapis.com/auth/compute', 'https://www.googleapis.com/auth/devstorage.read_only', 'https://www.googleapis.com/auth/logging.write', 'https://www.googleapis.com/auth/monitoring']` as the default value for `google_container_cluster/node_config/oauth_scopes`

2. Update the example in the document to:

```
resource "google_container_cluster" "primary" {
    name = "marcellus-wallace"
    zone = "us-central1-a"
    initial_node_count = 3

    master_auth {
        username = "mr.yoda"
        password = "adoy.rm"
    }

    node_config {
        oauth_scopes = [
            'https://www.googleapis.com/auth/compute', 
            'https://www.googleapis.com/auth/devstorage.read_only', 
            'https://www.googleapis.com/auth/logging.write', 
            'https://www.googleapis.com/auth/monitoring'        
        ]
    }
}
```

**Comment by lwander** [comment]
Good catch! I'm leaning towards updating the example in the documentation to make the behavior more transparent. What do you think, @sparkprime?

**Comment by sparkprime** [comment]
Either that or match the default scopes in google_compute_instance

**Comment by publysher** [comment]
After some thinking, I personally would prefer the documentation change over changing the defaults; in both cases, the example becomes consistent with `gcloud container clusters create` , but the example is more explicit. 

For example, if in a later phase I would like to add the scope 'https://www.googleapis.com/auth/ndev.clouddns.readwrite' , the approach with default scopes could prompt me to write:

```
...
node_config {
  oauth_scopes = [ 'https://www.googleapis.com/auth/ndev.clouddns.readwrite' ]
}
...
```

which is obviously not exactly what I intended. In the approach with the updated example, this would not happen.

**Comment by lwander** [comment]
I agree, but the issue with changing the `google_compute_instance` behavior is that it will break existing infrastructure. So we are left with either handling scopes inconsistently within the Google provider, or consistently having the problem you noted above.

**Comment by publysher** [comment]
I think we mean the same: changing the documentation seems to be better than changing the defaults. Apologies if I was unclear :)

**Comment by sparkprime** [comment]
So if gcloud container clusters create has no scopes by default, we should probably be consistent with that rather than consistent with ourselves.  Since gcloud has already made the choice to be inconsistent with itself and therefore we should be consistent with gcloud's inconsistency :)

**Comment by lwander** [comment]
Closing since we have addressed this in #3735
